flink-formats/flink-orc/src/main/java/org/apache/flink/orc/writer/OrcBulkWriter.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.orc.writer;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.serialization.BulkWriter;
+import org.apache.flink.orc.vector.Vectorizer;
+
+import org.apache.orc.OrcProto;
+import org.apache.orc.Writer;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * A {@link BulkWriter} implementation that writes data in ORC format.
+ *
+ * @param <T> The type of element written.
+ */
+@PublicEvolving
+public class OrcBulkWriter<T> implements BulkWriter<T> {
+
+	private final Writer writer;
+	private final Vectorizer<T> vectorizer;
+
+	private transient List<OrcProto.UserMetadataItem> userMetadata;
```

Comment:
I do not think this should be here. I was referring to the method 

```
/**
   * Add a row batch to the ORC file.
   * @param batch the rows to add
   */
  void addRowBatch(VectorizedRowBatch batch) throws IOException;
```

of the `Writer` and _not_ the `appendUserMetadata()`. The former seems to be aimed to be exposed to the user, while the latter does not (from the Javadoc).

In addition, this method seems to be intended to be called at any time, and not at the instantiation of the `OrcBulkWriter`. The user can add metadata at any point, they will be appended to a list, and the list will be appended to the output file when it is finally written. This is why in my branch I just set the `Writer` to the `vectorizer` and let the user use it whenever he/she wants. I still believe that that is the correct way, correct me if I am wrong.

Reply:
I'm assuming this is a typo and you meant `addUserMetadata` like the earlier comments and not `addRowBatch`.